### PR TITLE
fix: hover behaviour on table cells

### PR DESCRIPTION
**Pull request by AdityaPimpalkar**
fixes - #1508

https://github.com/twentyhq/twenty/assets/13139771/cf0d2054-6723-4c6a-b7bd-38112b88c8a5

### CODE REVIEW

File: front/src/modules/ui/table/components/EntityTableCell.tsx [hunk 1]
```diff
@@ -10,6 +11,14 @@ import { ColumnIndexContext } from '../contexts/ColumnIndexContext';
 import { GenericEditableCell } from '../editable-cell/components/GenericEditableCell';
 import { useCurrentRowSelected } from '../hooks/useCurrentRowSelected';
 
+const StyledCell = styled.div`
+  &:hover {
```
Comment (charlesBochet) [inline front/src/modules/ui/table/components/EntityTableCell.tsx:15]: Let's not introduce a new container, we could make this happen in EditableCell directly as we are already tracking the hover state. 
We can pass isHovered state from `EditableCell` to `EditableCellDisplay` (and `EditableCellDisplayContainer`) as prop. then in `EditableCellDisplayContainer` use it to display the same style as for `softFocus` case in `StyledEditableCellDisplayModeOuterContainer`

**Review by charlesBochet — COMMENTED** [review]
Thanks, left comment!

File: front/src/modules/ui/table/editable-cell/components/EditableCell.tsx [hunk 2]
```diff
@@ -84,6 +84,23 @@ export function EditableCell({
 
   const hasSoftFocus = useIsSoftFocusOnCurrentCell();
 
+  function EditButton() {
```
Comment (lucasbordeau) [inline front/src/modules/ui/table/editable-cell/components/EditableCell.tsx:87]: Components should be in a standalone file. Maybe `EditableCellEditButton`

File: front/src/modules/ui/table/components/EntityTableCell.tsx (hunk 1 above)
Comment (AdityaPimpalkar) [reply front/src/modules/ui/table/components/EntityTableCell.tsx]: Thanks!

File: front/src/modules/ui/table/editable-cell/components/EditableCell.tsx (hunk 2 above)
Comment (AdityaPimpalkar) [reply front/src/modules/ui/table/editable-cell/components/EditableCell.tsx]: Done!

**Review by charlesBochet — APPROVED** [review]

**Comment by charlesBochet** [comment]
Merging it, thank you!

**Pull request merged**